packages/ember-htmlbars/tests/helpers/each_test.js

```diff
@@ -1,617 +0,0 @@
-    });
-
-    templateMyView = compile('{{name}}');
-    lookup.MyView  = MyView = EmberView.extend({ template: templateMyView });
-    owner.register('view:my-view', MyView);
-
-    templateMyEmptyView = compile('I\'m empty');
-    lookup.MyEmptyView = MyEmptyView = EmberView.extend({
-      template: templateMyEmptyView
-    });
-    owner.register('view:my-empty-view', MyEmptyView);
-
-    runAppend(view);
-  },
-
-  teardown() {
-    runDestroy(owner);
-    runDestroy(view);
-    owner = view = null;
-
-    context.lookup = originalLookup;
-
-    resetKeyword('view', originalViewKeyword);
-  }
-});
-
-var assertHTML = function(view, expectedHTML) {
-  var html = view.$().html();
-
-  // IE 8 (and prior?) adds the \r\n
-  html = html.replace(/<script[^>]*><\/script>/ig, '').replace(/[\r\n]/g, '');
-
-  equal(html, expectedHTML);
-};
-
-QUnit.test('it renders the template for each item in an array', function() {
-  assertHTML(view, 'Steve HoltAnnabelle');
-});
-
-QUnit.test('it updates the view if an item is added', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-  });
-
-  assertHTML(view, 'Steve HoltAnnabelleTom Dale');
-});
-
-QUnit.test('it updates the view if an item is removed', function() {
-  run(function() {
-    people.removeAt(0);
-  });
-
-  assertHTML(view, 'Annabelle');
-});
-
-QUnit.test('it updates the view if an item is replaced', function() {
-  run(function() {
-    people.removeAt(0);
-    people.insertAt(0, { name: 'Kazuki' });
-  });
-
-  assertHTML(view, 'KazukiAnnabelle');
-});
-
-QUnit.test('can add and replace in the same runloop', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-    people.removeAt(0);
-    people.insertAt(0, { name: 'Kazuki' });
-  });
-
-  assertHTML(view, 'KazukiAnnabelleTom Dale');
-});
-
-QUnit.test('can add and replace the object before the add in the same runloop', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-    people.removeAt(1);
-    people.insertAt(1, { name: 'Kazuki' });
-  });
-
-  assertHTML(view, 'Steve HoltKazukiTom Dale');
-});
-
-QUnit.test('can add and replace complicatedly', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-    people.removeAt(1);
-    people.insertAt(1, { name: 'Kazuki' });
-    people.pushObject({ name: 'Firestone' });
-    people.pushObject({ name: 'McMunch' });
-    people.removeAt(3);
-  });
-
-  assertHTML(view, 'Steve HoltKazukiTom DaleMcMunch');
-});
-
-QUnit.test('can add and replace complicatedly harder', function() {
-  run(function() {
-    people.pushObject({ name: 'Tom Dale' });
-    people.removeAt(1);
-    people.insertAt(1, { name: 'Kazuki' });
-    people.pushObject({ name: 'Firestone' });
-    people.pushObject({ name: 'McMunch' });
-    people.removeAt(2);
-  });
-
-  assertHTML(view, 'Steve HoltKazukiFirestoneMcMunch');
-});
-
-QUnit.test('it does not mark each option tag as selected', function() {
```

Comment:
Same as above, but you may want to make your the 'U' part of your I-N-U-R do something more complicated like this.  The 'complicatedly harder' was added because 'complicatedly' didn't cover all of the diff-ing issues when childViews diff-ing was added to solve this issues way back when.